File: 백준/Silver/11724. 연결 요소의 개수/연결 요소의 개수.py
import sys
input = sys.stdin.readline
sys.setrecursionlimit(10**6)

def dfs(v):
    vst[v] = True
    for i in g[v]:
        if not vst[i]:
            dfs(i)

N, M = map(int, input().split())
g = [[] for i in range(N + 1)]
vst = [False] * (N + 1)
cnt = 0
for i in range(M):
    a, b = map(int, input().split())
    g[a].append(b)
    g[b].append(a)

for i in range(N):
    if not vst[i + 1]:
        dfs(i+1)
        cnt += 1
print(cnt)

File: 백준/Silver/11724. 연결 요소의 개수/연결 요소의 개수.cc
#include<iostream>
#include<queue>
using namespace std;

vector<int> vec[1001];
int vst[1001];
int n, m, cnt = 0;
void dfs(int v) {
    vst[v] = 1;
    for (int i = 0; i < vec[v].size(); i++) {
        int idx = vec[v][i];
        if (vst[idx] == 0) {
            dfs(idx);
        }
    }
}
int main() {
    ios::sync_with_stdio(false), cin.tie(NULL);

    cin >> n >> m;
    for (int i = 0; i < m; i++) {
        int a, b;
        cin >> a >> b;
        vec[a].push_back(b);
        vec[b].push_back(a);
    }
    for (int i = 1; i <= n; i++) {
        if (vst[i] == 0) {
            cnt++;
            dfs(i);
        }
    }
    cout << cnt << endl;
}
